### Read length distribution
Each of the read files contained reads of 15-75 nt in length. We extracted the read size distributions of the trimmed reads as follows:
```ShellSession
awk '{if(NR%4==2) print length($1)}' SequencingData_RWTH/smallRNAseq/Bgh-K1_haustoria-vesicles-epidermal_2019PietroSpanu/trimmed/RSB15248-1.trimmed.fastq | sort -n | uniq -c > RSB15248-1.sizeDistr.txt
```
This results in a two-column output, where column 1 contains the counts and column 2 the read size. To summarize the results in one table:
```ShellSession
paste <(awk '{print $2}' RSB15248-1.sizeDistr.txt) <(awk '{print $1}' RSB15248-1.sizeDistr.txt) <(awk '{print $1}' RSB15248-2.sizeDistr.txt) <(awk '{print $1}' RSB15248-3.sizeDistr.txt) <(awk '{print $1}' RSB15248-4.sizeDistr.txt) <(awk '{print $1}' RSB15248-5.sizeDistr.txt) <(awk '{print $1}' RSB15248-6.sizeDistr.txt) <(awk '{print $1}' RSB15248-7.sizeDistr.txt) <(awk '{print $1}' RSB15248-8.sizeDistr.txt) <(awk '{print $1}' RSB15248-9.sizeDistr.txt) <(awk '{print $1}' RSB15248-10.sizeDistr.txt) <(awk '{print $1}' RSB15248-11.sizeDistr.txt) <(awk '{print $1}' RSB15248-12.sizeDistr.txt) <(awk '{print $1}' RSB15248-13.sizeDistr.txt) <(awk '{print $1}' RSB15248-14.sizeDistr.txt) <(awk '{print $1}' RSB15248-15.sizeDistr.txt) <(awk '{print $1}' RSB15248-16.sizeDistr.txt) <(awk '{print $1}' RSB15248-17.sizeDistr.txt) <(awk '{print $1}' RSB15248-18.sizeDistr.txt) > sizeDistr.txt
```
Generating a table with read size distribution, which can be inspected by `head sizeDistr.txt`:
```
15	1380177	1290395	1796106	836532	653638	762148	363552	330822	300718	186332	292662	224531	271791	474893	202018	588802	707857	701715
16	4385732	3648088	3364706	924891	596408	882130	472160	477660	419417	240304	504547	289781	377443	415108	286773	1080844	1263246	1179919
17	1558281	1706594	2822221	1458984	946189	1408104	870096	834546	752348	353522	695557	339091	537506	583595	453019	1750946	2012774	1861221
18	1855285	1962225	2984519	2130960	1411695	2025815	1786815	1564398	1448955	503534	904296	436786	479231	642339	511473	2133160	2431974	2223099
19	3604061	3294956	3164676	2126402	1417759	2031843	1591491	1524530	1363873	506642	994414	428036	520634	622376	509871	2223685	2522453	2331232
20	2581484	2549389	2846196	1511700	1025126	1458101	869478	876975	805377	611789	1109874	455417	519656	621036	578042	2488148	2749977	2570071
```

### Extracting reads by read length
More detailed analysis of reads of specific lengths required extracting the reads of read length ranges. We used the following `Python` script for this: 
```Python

```

### Read mapping statistics by read length
We also generated statistics of the number of reads mapping to specific features in the two reference genomes, separated by read length. First, we created a read length index. 
```ShellSession
samtools faidx RSB15248-1.trimmed.fasta
awk '{print $1,$2}' RSB15248-1.trimmed.fasta.fai | sed 's/\:/\_/g' > RSB15248-1.trimmed.fasta.length
```
#### Overall read mapping to *Blumeria graminis* f.sp. *hordei* and *Hordeum vulgare*
We then used the BAM output files from `bowtie` mapping of short reads within the `ShortStack` pipeline. Mapping was done against the reference genomes of *Blumeria graminis* f.sp. *hordei* [(Frantzeskakis et al., 2018)](https://doi.org/10.1186/s12864-018-4750-6) and *Hordeum vulgare* [(Hordeum_vulgare.IBSC_v2)](http://plants.ensembl.org/Hordeum_vulgare/Info/Index), respectively. First, we determined the overall mapping to both genomes:
```ShellSession
# Blumeria graminis f.sp. hordei
samtools view -F 4 RSB15248-1.bgh_aligned.bam | awk '{print $1}' | sed 's/\:/\_/g' | awk '!x[$0]++' | grep -Fwf - RSB15248-1.trimmed.fasta.length | awk '{print $2}' | sort -n | uniq -c | awk '{print $1,$2}' > RSB15248-1.bgh_all_mapped.txt
# Hordeum vulgare
samtools view -F 4 merged_RSB15248-1.hvu_aligned.bam | awk '{print $1}' | sed 's/\:/\_/g' | awk '!x[$0]++' | grep -Fwf - RSB15248-1.trimmed.fasta.length | awk '{print $2}' | sort -n | uniq -c | awk '{print $1,$2}' > RSB15248-1.hvu_all_mapped.txt
```
We summarized the results from each sample into one document as follows: 
```ShellSession
# Blumeria graminis f.sp. hordei
paste RSB15248-1.bgh_all_mapped.txt RSB15248-2.bgh_all_mapped.txt RSB15248-3.bgh_all_mapped.txt RSB15248-4.bgh_all_mapped.txt RSB15248-5.bgh_all_mapped.txt RSB15248-6.bgh_all_mapped.txt RSB15248-7.bgh_all_mapped.txt RSB15248-8.bgh_all_mapped.txt RSB15248-9.bgh_all_mapped.txt RSB15248-10.bgh_all_mapped.txt RSB15248-11.bgh_all_mapped.txt RSB15248-12.bgh_all_mapped.txt RSB15248-13.bgh_all_mapped.txt RSB15248-14.bgh_all_mapped.txt RSB15248-15.bgh_all_mapped.txt RSB15248-16.bgh_all_mapped.txt RSB15248-17.bgh_all_mapped.txt RSB15248-18.bgh_all_mapped.txt | cut -f 1,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36 > bgh_all_mapped.txt
# Hordeum vulgare
paste RSB15248-1.hvu_all_mapped.txt RSB15248-2.hvu_all_mapped.txt RSB15248-3.hvu_all_mapped.txt RSB15248-4.hvu_all_mapped.txt RSB15248-5.hvu_all_mapped.txt RSB15248-6.hvu_all_mapped.txt RSB15248-7.hvu_all_mapped.txt RSB15248-8.hvu_all_mapped.txt RSB15248-9.hvu_all_mapped.txt RSB15248-10.hvu_all_mapped.txt RSB15248-11.hvu_all_mapped.txt RSB15248-12.hvu_all_mapped.txt RSB15248-13.hvu_all_mapped.txt RSB15248-14.hvu_all_mapped.txt RSB15248-15.hvu_all_mapped.txt RSB15248-16.hvu_all_mapped.txt RSB15248-17.hvu_all_mapped.txt RSB15248-18.hvu_all_mapped.txt | cut -f 1,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36 > hvu_all_mapped.txt
```

#### Feature-specific read mapping
In order to determine the number of reads of a given read length mapping to genomic features, i.e., transposable elements, coding genes, or milRNA genes, we first created necessary temporary files, i.e., the SAM header, and a non-repetitive text (SAM) file. 
```ShellSession
samtools view -F 4 RSB15248-1.bgh_aligned.bam | grep '@' > RSB15248-1.head # SAM header needed
samtools view -F 4 RSB15248-1.bgh_aligned.bam | awk '!visited[$1]++' | awk '{gsub(":","_",$1)}1' | sed 's/[[:space:]]/\t/g' > RSB15248-1.bgh_aligned.sam # temporary non-redundant read mapping file
```
Then, creating the read mapping SAM files for each read length separately:
```ShellSession
grep -Fwf <(awk '($2==15)' RSB15248-1.trimmed.fasta.length | awk '{print $1}') RSB15248-1.bgh_aligned.sam | cat RSB15248-1.head - > RSB15248-1_15.bgh.sam
...
grep -Fwf <(awk '($2==40)' RSB15248-1.trimmed.fasta.length | awk '{print $1}') RSB15248-1.bgh_aligned.sam | cat RSB15248-1.head - > RSB15248-1_40.bgh.sam
```
We then used `featureCounts` to determine the number of reads mapping to the respective features. We used the `GTF` annotation files for coding genes, milRNAs, and transposable elements, parsed to look as follows:
```
> head 

> head

> head

```
The counts were determined for RSB15248-1 (same procedure for the other 17 samples): 
```ShellSession
# mapped to milRNAs
featureCounts -T 2 -p -M --fraction -t nc_RNA -g ID -a Bgh_milRNA_v1.gff -o RSB15248-1.bgh-milRNA.txt RSB15248-1_15.bgh.sam RSB15248-1_16.bgh.sam RSB15248-1_17.bgh.sam RSB15248-1_18.bgh.sam RSB15248-1_19.bgh.sam RSB15248-1_20.bgh.sam RSB15248-1_21.bgh.sam RSB15248-1_22.bgh.sam RSB15248-1_23.bgh.sam RSB15248-1_24.bgh.sam RSB15248-1_25.bgh.sam RSB15248-1_26.bgh.sam RSB15248-1_27.bgh.sam RSB15248-1_28.bgh.sam RSB15248-1_29.bgh.sam RSB15248-1_30.bgh.sam RSB15248-1_31.bgh.sam RSB15248-1_32.bgh.sam RSB15248-1_33.bgh.sam RSB15248-1_34.bgh.sam RSB15248-1_35.bgh.sam RSB15248-1_36.bgh.sam RSB15248-1_37.bgh.sam RSB15248-1_38.bgh.sam RSB15248-1_39.bgh.sam RSB15248-1_40.bgh.sam
# mapped to coding genes
featureCounts -T 2 -p -M --fraction -t mRNA -g ID -a bgh_dh14_v4-2.gff -o RSB15248-1.bgh-gene.txt RSB15248-1_15.bgh.sam RSB15248-1_16.bgh.sam RSB15248-1_17.bgh.sam RSB15248-1_18.bgh.sam RSB15248-1_19.bgh.sam RSB15248-1_20.bgh.sam RSB15248-1_21.bgh.sam RSB15248-1_22.bgh.sam RSB15248-1_23.bgh.sam RSB15248-1_24.bgh.sam RSB15248-1_25.bgh.sam RSB15248-1_26.bgh.sam RSB15248-1_27.bgh.sam RSB15248-1_28.bgh.sam RSB15248-1_29.bgh.sam RSB15248-1_30.bgh.sam RSB15248-1_31.bgh.sam RSB15248-1_32.bgh.sam RSB15248-1_33.bgh.sam RSB15248-1_34.bgh.sam RSB15248-1_35.bgh.sam RSB15248-1_36.bgh.sam RSB15248-1_37.bgh.sam RSB15248-1_38.bgh.sam RSB15248-1_39.bgh.sam RSB15248-1_40.bgh.sam
# mapped to transposable elements
featureCounts -T 2 -p -M --fraction -t gene -g gene_id -a bgh_dh14_v4.fa.out.gtf -o RSB15248-1.bgh-TE.txt RSB15248-1_15.bgh.sam RSB15248-1_16.bgh.sam RSB15248-1_17.bgh.sam RSB15248-1_18.bgh.sam RSB15248-1_19.bgh.sam RSB15248-1_20.bgh.sam RSB15248-1_21.bgh.sam RSB15248-1_22.bgh.sam RSB15248-1_23.bgh.sam RSB15248-1_24.bgh.sam RSB15248-1_25.bgh.sam RSB15248-1_26.bgh.sam RSB15248-1_27.bgh.sam RSB15248-1_28.bgh.sam RSB15248-1_29.bgh.sam RSB15248-1_30.bgh.sam RSB15248-1_31.bgh.sam RSB15248-1_32.bgh.sam RSB15248-1_33.bgh.sam RSB15248-1_34.bgh.sam RSB15248-1_35.bgh.sam RSB15248-1_36.bgh.sam RSB15248-1_37.bgh.sam RSB15248-1_38.bgh.sam RSB15248-1_39.bgh.sam RSB15248-1_40.bgh.sam
```
