### Extracting reads by read length
More detailed analysis of reads of specific lengths required extracting the reads of read length ranges. First, we parsed FASTQ to FASTA.
```ShellSession
# generating FASTA read file
seqkit fq2fa RSB15248-1.trimmed.fastq -o RSB15248-1.trimmed.fa
# simplifying FASTA headers
sed 's/[[:space:]].*//g' RSB15248-1.trimmed.fa | sed 's/\:/\_/g' > RSB15248-1.trimmed.fasta
```
Then, used the following `Python` script for extracting reads of specific length integers. 
```Python
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Wed Jul  7 10:20:18 2021

@author: mansisingh
"""

#import packages 
from Bio import SeqIO

#to make the text file
text_file1 = open("RSB15248-1_2122.fa","w")
text_file2 = open("RSB15248-1_3132.fa","w")

#select sequences based on their length 
fasta_sequences = SeqIO.parse(open('RSB15248-1.trimmed.fasta'),'fasta')
for fasta in fasta_sequences:
        name, sequence = fasta.id, str(fasta.seq)
        if len(sequence) == 21 or len(sequence) == 22:
            text_file1.write(">" + name + "\n" + sequence + "\n")
            
        if len(sequence) == 31 or len(sequence) == 32:
          	text_file2.write(">" + name + "\n" + sequence + "\n")

text_file1.close()
text_file2.close()
```
The process was repeated for all samples, RSB15248-1 to RSB15248-18. Finally, we used `cat` to combine the replicate files.
```ShellSession
# reads 21-22 nt length
cat RSB15248-1_2122.fasta RSB15248-2_2122.fasta RSB15248-3_2122.fasta > Epi2122.fasta
cat RSB15248-4_2122.fasta RSB15248-5_2122.fasta RSB15248-6_2122.fasta > Hau2122.fasta
cat RSB15248-7_2122.fasta RSB15248-8_2122.fasta RSB15248-9_2122.fasta > P402122.fasta
cat RSB15248-10_2122.fasta RSB15248-11_2122.fasta RSB15248-15_2122.fasta > EVplus2122.fasta
cat RSB15248-12_2122.fasta RSB15248-13_2122.fasta RSB15248-14_2122.fasta > EVminus2122.fasta
cat RSB15248-16_2122.fasta RSB15248-17_2122.fasta RSB15248-18_2122.fasta > Myc2122.fasta
# reads 31-32 nt length
cat RSB15248-1_3132.fasta RSB15248-2_3132.fasta RSB15248-3_3132.fasta > Epi3132.fasta
cat RSB15248-4_3132.fasta RSB15248-5_3132.fasta RSB15248-6_3132.fasta > Hau3132.fasta
cat RSB15248-7_3132.fasta RSB15248-8_3132.fasta RSB15248-9_3132.fasta > P403132.fasta
cat RSB15248-10_3132.fasta RSB15248-11_3132.fasta RSB15248-15_3132.fasta > EVplus3132.fasta
cat RSB15248-12_3132.fasta RSB15248-13_3132.fasta RSB15248-14_3132.fasta > EVminus3132.fasta
cat RSB15248-16_3132.fasta RSB15248-17_3132.fasta RSB15248-18_3132.fasta > Myc3132.fasta
```

### Deduplicating reads
For better handling of the file size by collapsing identical reads into a single entry with a count value, we used `clumpify.sh` from the `BBmap` utility. We ran this line for all FASTA files above. 
```ShellSession
clumpify.sh in=Myc3132.fasta out=Myc3132_dedup.fa dedupe addcount
```
The output looks like this, containing the copy number of reads in the `copies=` suffix of the read name. 
```ShellSession
$ grep '=' Myc3132_dedup.fa | head 
>NB551536_74_H7CVKBGXF_4_12401_6241_1695_copies=2
>NB551536_74_H7CVKBGXF_4_13402_17980_20149_copies=2
>NB551536_74_H7CVKBGXF_1_23108_14983_10310_copies=10
>NB551536_74_H7CVKBGXF_1_11203_23613_16582_copies=18
>NB551536_74_H7CVKBGXF_1_21106_13537_13755_copies=3
>NB551536_74_H7CVKBGXF_1_12105_17811_18613_copies=5
```

### Obtaining RFAM databases 
We downloaded the FASTA files containing conserved ribosomal RNA sequences from [https://rfam.xfam.org](http://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/fasta_files/)
```ShellSession
wget http://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/fasta_files/RF00001.fa.gz
wget http://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/fasta_files/RF00002.fa.gz
wget http://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/fasta_files/RF00005.fa.gz
wget http://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/fasta_files/RF01960.fa.gz
wget http://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/fasta_files/RF02543.fa.gz
```

### Creating `MMSeqs2` databases
To run `MMSeqs2`, we first created the required `MMSeqs2` databases, which the tool requires for sequence alignment search. 
```ShellSession
# RFAM MMSeqs2 databases
mmseqs createdb RF00001.fa rf00001db
mmseqs createindex rf00001db --search-type 3
mmseqs createdb RF00002.fa rf00002db
mmseqs createindex rf00002db --search-type 3
mmseqs createdb RF00005.fa rf00005db
mmseqs createindex rf00005db --search-type 3
mmseqs createdb RF01960.fa rf01960db
mmseqs createindex rf01960db --search-type 3
mmseqs createdb RF02543.fa rf02543db
mmseqs createindex rf02543db --search-type 3
# EV minus 31-32 nt
mmseqs createdb EVminus3132_dedup.fa EVminusDB 
mmseqs createindex EVminusDB tmp --search-type 3
# EV plus 31-32 nt
mmseqs createdb EVplus3132_dedup.fa EVplusDB 
mmseqs createindex EVplusDB tmp --search-type 3
# Epi 31-32 nt
mmseqs createdb Epi3132_dedup.fa EpiDB 
mmseqs createindex EpiDB tmp --search-type 3
# Myc 31-32 nt
mmseqs createdb Myc3132_dedup.fa MycDB 
mmseqs createindex MycDB tmp --search-type 3
# P40 31-32 nt
mmseqs createdb P403132_dedup.fa P40DB 
mmseqs createindex P40DB tmp --search-type 3
# Hau 31-32 nt
mmseqs createdb Hau3132_dedup.fa HauDB 
mmseqs createindex HauDB tmp --search-type 3
```

### Running `MMSeqs2` against RFAM databases
With the databases set up, we ran `MMSeqs2`. The following scripts were run for all sample and RFAM databases above. 
```ShellSession
# run mmseqs
mmseqs search --threads 48 --search-type 3 -s 4.0 MycDB rf00001db Myc_rf00001DB tmp
# create readable result table
mmseqs convertalis MycDB rf00001db MycRfamDB Myc_rf00001DB.m8
# remove large search database to free up space
rm Myc_rf00001DB*
```
Because reads can have several hits in the database, we reduced the table to retain only the best hit for each read. This reduced the file size and enabled downstream analysis. The script first extracts reads that align with *Blumeria graminis* f.sp. *hordei* (CAUH01 genome accession in RFAM) or *Hordeum vulgare* (CAJW01 genome accession in RFAM). 
```ShellSession
# first for ribosomal RNAs in RF00001, RF00002, RF01960, and RF02543 in B. graminis f.sp. hordei or H. vulgare
cat <(grep CAUH01 Myc_rf00001.m8) <(grep CAJW Myc_rf00001.m8) <(grep CAUH01 Myc_rf00002.m8) <(grep CAJW Myc_rf00002.m8) <(grep CAUH01 Myc_rf01960.m8) <(grep CAJW Myc_rf01960.m8) <(grep CAUH01 Myc_rf02543.m8) <(grep CAJW Myc_rf02543.m8) | sort -k1,1 -k11r,11 > Myc_rRNA_hvubgh.m8;
awk '!visited[$1]++' Myc_rRNA_hvubgh.m8 > Myc_rRNA_hvubgh.besthit.m8
# extracting additional reads aligning with ribosomal RNA from other organisms
awk '{print $1}' Myc_rRNA_hvubgh.besthit.m8 | grep -Fvf - <(cat Myc_rf00001.m8 Myc_rf00002.m8 Myc_rf01960.m8 Myc_rf02543.m8) | sort -k1,1 -k11r,11 | awk '!visited[$1]++' > Myc_rRNA_other.besthit.m8
# then extracting reads aligning with tRNA from B. graminis f.sp. hordei or H. vulgare
cat <(grep CAUH01 Myc_rf00005.m8) <(grep CAJW Myc_rf00005.m8) | sort -k1,1 -k11r,11 > Myc_tRNA_hvubgh.m8
awk '!visited[$1]++' Myc_tRNA_hvubgh.m8 > Myc_tRNA_hvubgh.besthit.m8
```

### Extracting quantitative information from besthit tables
We used a combination of scripts to extract read counts aligning with the respective RFAM family, separately for reads from *Hordeum vulgare*, *Blumeria graminis* f.sp. *hordei*, and others. 
```ShellSession
grep CAJW Myc_rfam_hvubgh_besthit.m8 | grep rf00001 | sed 's/\=/\t/g' | awk '{SUM+=$2}END{print SUM}'
grep CAJW Myc_rfam_hvubgh_besthit.m8 | grep rf00002 | sed 's/\=/\t/g' | awk '{SUM+=$2}END{print SUM}'
grep CAJW Myc_rfam_hvubgh_besthit.m8 | grep rf01960 | sed 's/\=/\t/g' | awk '{SUM+=$2}END{print SUM}'
grep CAJW Myc_rfam_hvubgh_besthit.m8 | grep rf02543 | sed 's/\=/\t/g' | awk '{SUM+=$2}END{print SUM}'
grep CAJW Myc_rfam_hvubgh_besthit.m8 | grep rf00001 | grep -v copies | awk '!x[$0]++' | awk 'BEGIN{i=0}{i++;}END{print i}'
grep CAJW Myc_rfam_hvubgh_besthit.m8 | grep rf00002 | grep -v copies | awk '!x[$0]++' | awk 'BEGIN{i=0}{i++;}END{print i}'
grep CAJW Myc_rfam_hvubgh_besthit.m8 | grep rf01960 | grep -v copies | awk '!x[$0]++' | awk 'BEGIN{i=0}{i++;}END{print i}'
grep CAJW Myc_rfam_hvubgh_besthit.m8 | grep rf02543 | grep -v copies | awk '!x[$0]++' | awk 'BEGIN{i=0}{i++;}END{print i}'
grep CAUH Myc_rfam_hvubgh_besthit.m8 | grep rf00001 | sed 's/\=/\t/g' | awk '{SUM+=$2}END{print SUM}'
grep CAUH Myc_rfam_hvubgh_besthit.m8 | grep rf00002 | sed 's/\=/\t/g' | awk '{SUM+=$2}END{print SUM}'
grep CAUH Myc_rfam_hvubgh_besthit.m8 | grep rf01960 | sed 's/\=/\t/g' | awk '{SUM+=$2}END{print SUM}'
grep CAUH Myc_rfam_hvubgh_besthit.m8 | grep rf02543 | sed 's/\=/\t/g' | awk '{SUM+=$2}END{print SUM}'
grep CAUH Myc_rfam_hvubgh_besthit.m8 | grep rf00001 | grep -v copies | awk '!x[$0]++' | awk 'BEGIN{i=0}{i++;}END{print i}'
grep CAUH Myc_rfam_hvubgh_besthit.m8 | grep rf00002 | grep -v copies | awk '!x[$0]++' | awk 'BEGIN{i=0}{i++;}END{print i}'
grep CAUH Myc_rfam_hvubgh_besthit.m8 | grep rf01960 | grep -v copies | awk '!x[$0]++' | awk 'BEGIN{i=0}{i++;}END{print i}'
grep CAUH Myc_rfam_hvubgh_besthit.m8 | grep rf02543 | grep -v copies | awk '!x[$0]++' | awk 'BEGIN{i=0}{i++;}END{print i}'
```
