#### Obtaining RFAM databases 
For a quick overview over some genome statistics, I used `assembly-stats`:
```ShellSession
assembly-stats Ltaurica_exCapsicumHu_v1.fa
```
The output includes N50, N90, and other information: 
```
sum = 187262869, n = 23599, ave = 7935.20, largest = 99596
N50 = 13899, n = 4047
N60 = 10976, n = 5564
N70 = 8378, n = 7510
N80 = 5893, n = 10162
N90 = 3522, n = 14214
N100 = 1000, n = 23599
N_count = 2326588
Gaps = 84968
```

#### Creating 'MMSeqs2' databases
A very useful measure of genome quality is to use `BUSCO v3`. This algorithm searches for core conserved gene sets of lineages to assess the completeness of a given genome or proteome.  

```ShellSession
python $HOME/miniconda3/bin/BUSCO.py -f --out Ltaurica_v1.busco --in Ltaurica_exCapsicumHu_v1.fasta --lineage ~/database/BUSCO/ascomycota_odb9 -m genome
```

The summary output of `BUSCO` reveals the number of complete, fragmented, duplicated, and missing core genes of the selected lineage, here ascomycota_odb9.
```
C:85.0%[S:85.0%,D:0.0%],F:1.3%,M:13.7%,n:1315

	1118	Complete BUSCOs (C)
	1118	Complete and single-copy BUSCOs (S)
	0	Complete and duplicated BUSCOs (D)
	17	Fragmented BUSCOs (F)
	180	Missing BUSCOs (M)
	1315	Total BUSCO groups searched
```
