### Extracting reads by read length
More detailed analysis of reads of specific lengths required extracting the reads of read length ranges. First, we parsed FASTQ to FASTA.
```ShellSession
#generating FASTA read file
seqkit fq2fa RSB15248-1.trimmed.fastq -o RSB15248-1.trimmed.fa
#simplifying FASTA headers
sed 's/[[:space:]].*//g' RSB15248-1.trimmed.fa | sed 's/\:/\_/g' > RSB15248-1.trimmed.fasta
```
Then, used the following `Python` script for extracting reads of specific length integers. 
```Python
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Wed Jul  7 10:20:18 2021

@author: mansisingh
"""

#import packages 
from Bio import SeqIO

#to make the text file
text_file1 = open("RSB15248-1_2122.fa","w")
text_file2 = open("RSB15248-1_3132.fa","w")

#select sequences based on their length 
fasta_sequences = SeqIO.parse(open('RSB15248-1.trimmed.fasta'),'fasta')
for fasta in fasta_sequences:
        name, sequence = fasta.id, str(fasta.seq)
        if len(sequence) == 21 or len(sequence) == 22:
            text_file1.write(">" + name + "\n" + sequence + "\n")
            
        if len(sequence) == 31 or len(sequence) == 32:
          	text_file2.write(">" + name + "\n" + sequence + "\n")

text_file1.close()
text_file2.close()
```
The process was repeated for all samples, RSB15248-1 to RSB15248-18. Finally, we used `cat` to combine the replicate files.
```ShellSession
# reads 21-22 nt length
cat RSB15248-1_2122.fasta RSB15248-2_2122.fasta RSB15248-3_2122.fasta > Epi2122.fasta
cat RSB15248-4_2122.fasta RSB15248-5_2122.fasta RSB15248-6_2122.fasta > Hau2122.fasta
cat RSB15248-7_2122.fasta RSB15248-8_2122.fasta RSB15248-9_2122.fasta > P402122.fasta
cat RSB15248-10_2122.fasta RSB15248-11_2122.fasta RSB15248-15_2122.fasta > EVplus2122.fasta
cat RSB15248-12_2122.fasta RSB15248-13_2122.fasta RSB15248-14_2122.fasta > EVminus2122.fasta
cat RSB15248-16_2122.fasta RSB15248-17_2122.fasta RSB15248-18_2122.fasta > Myc2122.fasta
# reads 31-32 nt length
cat RSB15248-1_3132.fasta RSB15248-2_3132.fasta RSB15248-3_3132.fasta > Epi3132.fasta
cat RSB15248-4_3132.fasta RSB15248-5_3132.fasta RSB15248-6_3132.fasta > Hau3132.fasta
cat RSB15248-7_3132.fasta RSB15248-8_3132.fasta RSB15248-9_3132.fasta > P403132.fasta
cat RSB15248-10_3132.fasta RSB15248-11_3132.fasta RSB15248-15_3132.fasta > EVplus3132.fasta
cat RSB15248-12_3132.fasta RSB15248-13_3132.fasta RSB15248-14_3132.fasta > EVminus3132.fasta
cat RSB15248-16_3132.fasta RSB15248-17_3132.fasta RSB15248-18_3132.fasta > Myc3132.fasta
```

### Obtaining RFAM databases 
For a quick overview over some genome statistics, I used `assembly-stats`:
```ShellSession
assembly-stats Ltaurica_exCapsicumHu_v1.fa
```

#### Creating 'MMSeqs2' databases
A very useful measure of genome quality is to use `BUSCO v3`. This algorithm searches for core conserved gene sets of lineages to assess the completeness of a given genome or proteome.  

```ShellSession
python $HOME/miniconda3/bin/BUSCO.py -f --out Ltaurica_v1.busco --in Ltaurica_exCapsicumHu_v1.fasta --lineage ~/database/BUSCO/ascomycota_odb9 -m genome
```
