### Running `ShortStack` to discover candidate milRNAs
We used `ShortStack` and the trimmed sRNA-seq read files as input to predict milRNAs in *Blumeria graminis* f.sp. *hordei* and *Hordeum vulgare*. In case of *Blumeria graminis* f.sp. *hordei*, we ran the code using the full genome assembly. The code was run for files RSB15248-1 through RSB15248-18, generating a total of 18 `ShortStack` outputs. 
```ShellSession
ShortStack -dicermin 18 -mincov 2 --readfile RSB15248-1.ftrimmed.fastq --genomefile bgh_dh14_v4.fa
```

For *Hordeum vulgare*, we ran the `ShortStack` pipeline for each of the 8 *Hordeum vulgare* chromosomes separetely, since the genome was too large to process as a whole. The codes were run for files RSB15248-1 through RSB15248-18, generating a total of 144 `ShortStack` outputs.
```ShellSession
ShortStack -dicermin 18 -mincov 2 --readfile RSB15248-1.ftrimmed.fastq --genomefile Hordeum_vulgare.IBSC_v2.chr1H.fa
ShortStack -dicermin 18 -mincov 2 --readfile RSB15248-1.ftrimmed.fastq --genomefile Hordeum_vulgare.IBSC_v2.chr2H.fa
ShortStack -dicermin 18 -mincov 2 --readfile RSB15248-1.ftrimmed.fastq --genomefile Hordeum_vulgare.IBSC_v2.chr3H.fa
ShortStack -dicermin 18 -mincov 2 --readfile RSB15248-1.ftrimmed.fastq --genomefile Hordeum_vulgare.IBSC_v2.chr4H.fa
ShortStack -dicermin 18 -mincov 2 --readfile RSB15248-1.ftrimmed.fastq --genomefile Hordeum_vulgare.IBSC_v2.chr5H.fa
ShortStack -dicermin 18 -mincov 2 --readfile RSB15248-1.ftrimmed.fastq --genomefile Hordeum_vulgare.IBSC_v2.chr6H.fa
ShortStack -dicermin 18 -mincov 2 --readfile RSB15248-1.ftrimmed.fastq --genomefile Hordeum_vulgare.IBSC_v2.chr7H.fa
ShortStack -dicermin 18 -mincov 2 --readfile RSB15248-1.ftrimmed.fastq --genomefile Hordeum_vulgare.IBSC_v2.chrUn.fa
```

### Parsing tables for obtaining non-redundant milRNA loci
