Proper gene annotation of a newly sequenced genome requires masking of repetitive elements (transposons), then automated gene prediction, followed by manual curation. Tools for gene prediction include `Augustus` and `BRAKER`. 

#### Running `ShortStack`
This step is important to label transposable elements that might mess up the annotation, and additionally can provide information about the nature and distribution of different types of transposons. In the following line, the analysis was done for a fungal species. `RepeatMasker -h` provides a list of all functions. 
```ShellSession
RepeatMasker -species fungi -small -nolow -dir RepMasker Ltaurica_exCapsicumHu_v1.fa
```
`-nolow` prevents masking of low complexity and simple repeats, `-small` returns masked sequence in lower case (standard is N). The output will be a fa.masked fasta file. 

#### Parsing tables for obtaining non-redundant milRNA loci
A pretty extensive tutorial on the usage of `Augustus` can be found here: http://bioinf.uni-greifswald.de/augustus/binaries/tutorial/. Most `perl`-scripts are provided in the scripts/ folder of `Augustus`. 

   - **training `Augustus` with gene models**

