## Extracting read counts of milRNA gene loci
First, we needed to assing read counts to milRNA loci in *Blumeria graminis* f.sp. *hordei* and *Hordeum vulgare*. We used `featureCounts` for this, as before. 
```ShellSession
# Creating table of read counts for B. graminis f.sp. hordei
featureCounts -T 2 -p -M --fraction -t nc_RNA -g ID -a Bgh_milRNA_v1.gff -o counts.bgh-milRNA.txt RSB15248-1.bgh_aligned.bam RSB15248-2.bgh_aligned.bam RSB15248-3.bgh_aligned.bam RSB15248-4.bgh_aligned.bam RSB15248-5.bgh_aligned.bam RSB15248-6.bgh_aligned.bam RSB15248-7.bgh_aligned.bam RSB15248-8.bgh_aligned.bam RSB15248-9.bgh_aligned.bam RSB15248-10.bgh_aligned.bam RSB15248-11.bgh_aligned.bam RSB15248-12.bgh_aligned.bam RSB15248-13.bgh_aligned.bam RSB15248-14.bgh_aligned.bam RSB15248-15.bgh_aligned.bam RSB15248-16.bgh_aligned.bam RSB15248-17.bgh_aligned.bam RSB15248-18.bgh_aligned.bam
# Creating table of read counts for H. vulgare
featureCounts -T 8 -p -M --fraction -t nc_RNA -g ID -a Hvu_milRNA_v1.gff -o counts.hvu-milRNA.txt RSB15248-1_merged.hvu_aligned.bam RSB15248-2_merged.hvu_aligned.bam RSB15248-3_merged.hvu_aligned.bam RSB15248-4_merged.hvu_aligned.bam RSB15248-5_merged.hvu_aligned.bam RSB15248-6_merged.hvu_aligned.bam RSB15248-7_merged.hvu_aligned.bam RSB15248-8_merged.hvu_aligned.bam RSB15248-9_merged.hvu_aligned.bam RSB15248-10_merged.hvu_aligned.bam RSB15248-11_merged.hvu_aligned.bam RSB15248-12_merged.hvu_aligned.bam RSB15248-13_merged.hvu_aligned.bam RSB15248-14_merged.hvu_aligned.bam RSB15248-15_merged.hvu_aligned.bam RSB15248-16_merged.hvu_aligned.bam RSB15248-17_merged.hvu_aligned.bam RSB15248-18_merged.hvu_aligned.bam
```

## Parsing read count table and filtering

```R
counts <- read.table("counts.bgh-milRNA.txt", header=TRUE, sep=" ")
names(counts) <- c("geneID","length",
                "RSB15248-1", "RSB15248-2", "RSB15248-3", "RSB15248-4", "RSB15248-5", "RSB15248-6",
                "RSB15248-7", "RSB15248-8", "RSB15248-9", "RSB15248-10", "RSB15248-11", "RSB15248-12",
                "RSB15248-13", "RSB15248-14", "RSB15248-15", "RSB15248-16", "RSB15248-17", "RSB15248-18")
# Function TPM calculation
tpm <- function(counts, lengths) {
  rate <- counts / lengths
  rate / sum(rate) * 1e6
}
# Parse data
genes <- data.frame(counts[1:2])
counts <- counts
countsTable <- as.matrix(counts[,3:20])
rownames(countsTable) <- counts[,1]
# Calculate TPM
tpm <- apply(countsTable, 2, function(x) tpm(x, genes$length))
rownames(tpm) <- counts[,1]
# Checking that codes ran correctly
colSums(tpm) #should be equal
colMeans(tpm) #should be equal
```

## Clustering by NMDS

## Clustering by MDS

## Clustering by PCA
