## Extracting read counts of milRNA gene loci
First, we needed to assing read counts to milRNA loci in *Blumeria graminis* f.sp. *hordei* and *Hordeum vulgare*. We used `featureCounts` for this, as before. 
```ShellSession
# Creating table of read counts for B. graminis f.sp. hordei
featureCounts -T 2 -p -M --fraction -t nc_RNA -g ID -a Bgh_milRNA_v1.gff -o counts.bgh-milRNA.txt RSB15248-1.bgh_aligned.bam RSB15248-2.bgh_aligned.bam RSB15248-3.bgh_aligned.bam RSB15248-4.bgh_aligned.bam RSB15248-5.bgh_aligned.bam RSB15248-6.bgh_aligned.bam RSB15248-7.bgh_aligned.bam RSB15248-8.bgh_aligned.bam RSB15248-9.bgh_aligned.bam RSB15248-10.bgh_aligned.bam RSB15248-11.bgh_aligned.bam RSB15248-12.bgh_aligned.bam RSB15248-13.bgh_aligned.bam RSB15248-14.bgh_aligned.bam RSB15248-15.bgh_aligned.bam RSB15248-16.bgh_aligned.bam RSB15248-17.bgh_aligned.bam RSB15248-18.bgh_aligned.bam
# Creating table of read counts for H. vulgare
featureCounts -T 8 -p -M --fraction -t nc_RNA -g ID -a Hvu_milRNA_v1.gff -o counts.hvu-milRNA.txt RSB15248-1_merged.hvu_aligned.bam RSB15248-2_merged.hvu_aligned.bam RSB15248-3_merged.hvu_aligned.bam RSB15248-4_merged.hvu_aligned.bam RSB15248-5_merged.hvu_aligned.bam RSB15248-6_merged.hvu_aligned.bam RSB15248-7_merged.hvu_aligned.bam RSB15248-8_merged.hvu_aligned.bam RSB15248-9_merged.hvu_aligned.bam RSB15248-10_merged.hvu_aligned.bam RSB15248-11_merged.hvu_aligned.bam RSB15248-12_merged.hvu_aligned.bam RSB15248-13_merged.hvu_aligned.bam RSB15248-14_merged.hvu_aligned.bam RSB15248-15_merged.hvu_aligned.bam RSB15248-16_merged.hvu_aligned.bam RSB15248-17_merged.hvu_aligned.bam RSB15248-18_merged.hvu_aligned.bam
```

## Parsing read count table and filtering
We first created read count and TPM tables for downstream analysis in `R`. The codes are shown for *Blumeria graminis* f.sp. *hordei*, and were applied identically to *Hordeum vulgare*.
```R
counts <- read.table("counts.bgh-milRNA.txt", header=TRUE, sep=" ")
names(counts) <- c("geneID","length",
                "RSB15248-1", "RSB15248-2", "RSB15248-3", "RSB15248-4", "RSB15248-5", "RSB15248-6",
                "RSB15248-7", "RSB15248-8", "RSB15248-9", "RSB15248-10", "RSB15248-11", "RSB15248-12",
                "RSB15248-13", "RSB15248-14", "RSB15248-15", "RSB15248-16", "RSB15248-17", "RSB15248-18")
# Function TPM calculation
tpm <- function(counts, lengths) {
  rate <- counts / lengths
  rate / sum(rate) * 1e6
}
# Parse data
genes <- data.frame(counts[1:2])
counts <- counts
countsTable <- as.matrix(counts[,3:20])
rownames(countsTable) <- counts[,1]
# Calculate TPM
tpm <- apply(countsTable, 2, function(x) tpm(x, genes$length))
rownames(tpm) <- counts[,1]
# Checking that codes ran correctly
colSums(tpm) #should be equal
colMeans(tpm) #should be equal
```
We also calculated average and median TPM values for each condition, i.e., Myc, Epi, Hau, P40, EVplus, and EVminus.
```R
# Calculations of averages
Epi <- rowMeans(data.frame(tpm[,1],tpm[,2],tpm[,3]), na.rm = FALSE, dims = 1)
Hau <- rowMeans(data.frame(tpm[,4],tpm[,5],tpm[,6]), na.rm = FALSE, dims = 1)
P40 <- rowMeans(data.frame(tpm[,7],tpm[,8],tpm[,9]), na.rm = FALSE, dims = 1)
EVplus <- rowMeans(data.frame(tpm[,10],tpm[,11],tpm[,15]), na.rm = FALSE, dims = 1)
EVminus <- rowMeans(data.frame(tpm[,12],tpm[,13],tpm[,14]), na.rm = FALSE, dims = 1)
Myc <- rowMeans(data.frame(tpm[,16],tpm[,17],tpm[,18]), na.rm = FALSE, dims = 1)
# Combine
tpm.avg <- data.frame(Myc,Epi,Hau,P40,EVplus,EVminus)
# Calculations of medians
Epi <- apply(data.frame(tpm[,1],tpm[,2],tpm[,3]), 1, median)
Hau <- apply(data.frame(tpm[,4],tpm[,5],tpm[,6]), 1, median)
P40 <- apply(data.frame(tpm[,7],tpm[,8],tpm[,9]), 1, median)
EVplus <- apply(data.frame(tpm[,10],tpm[,11],tpm[,15]), 1, median)
EVminus <- apply(data.frame(tpm[,12],tpm[,13],tpm[,14]), 1, median)
Myc <- apply(data.frame(tpm[,16],tpm[,17],tpm[,18]), 1, median)
# Combine
tpm.median <- data.frame(Myc,Epi,Hau,P40,EVplus,EVminus)
head(tpm.median)
```
Filtering can be done using a TPM cut-off value of `TPM < 1`. 
```R
# Removing non-expressors using median values
maxv <- apply(tpm.median[,1:6], 1, max)
minv <- apply(tpm.median[,1:6], 1, min)

med.maxv <- data.frame(tpm.median[1:6], minv, maxv)
cond <- med.maxv$maxv < 1
tpm.filter1 <- med.maxv[!cond,]
tpm.filter1 <- data.frame(rownames(tpm.filter1), tpm.filter1)
names(tpm.filter1) <- c("geneID","Myc","Epi","Hau","P40","EVplus","EVminus", "minv", "maxv")
```

## Clustering by Nonmetric dimensional scaling (NMDS)
Non-metric Multi-dimensional Scaling (NMDS) condenses information from multidimensional data into a 2-dimensional representation or ordination. In this ordination, the closer two points are, the more similar the corresponding samples are with respect to the variables that went into calculating the NMDS plot. NMDS attempts to represent the pairwise dissimilarity between objects in a low-dimensional space.
Loading required packages.
```R
require(vegan)
```
Importing data for analysis. 
```R
# sample info
info <- read.table("sample_info.tsv", header = T, row.names = 1, sep = "\t")
groups <- info$Group
# import data
counts <- read.table("counts.bgh-milRNA.txt", header=TRUE, sep=" ")
names(counts) <- c("geneID","length",
                "RSB15248-1", "RSB15248-2", "RSB15248-3", "RSB15248-4", "RSB15248-5", "RSB15248-6",
                "RSB15248-7", "RSB15248-8", "RSB15248-9", "RSB15248-10", "RSB15248-11", "RSB15248-12",
                "RSB15248-13", "RSB15248-14", "RSB15248-15", "RSB15248-16", "RSB15248-17", "RSB15248-18")
tcounts <- t(counts)
```

## Clustering by PCA


## Hierarchical clustering by PCC

