## Filtering out non-expressed milRNA genes

## Differential expression analysis with `DESeq2`
Loading required packages. 
```R
require(Biobase)
require(DESeq2)
require(apeglm)
require(vsn)
require(ggplot2)
require(IHW)
require(NMF)
```
### Read count normalization
First, we normalized read counts across samples with the `DESeq2` method for normalization. The table `conditions.txt` looked as follows:
```
sample  condition
RSB15248-1	Epi
RSB15248-2	Epi
RSB15248-3	Epi
RSB15248-4	Hau
RSB15248-5	Hau
RSB15248-6	Hau
RSB15248-7	P40
RSB15248-8	P40
RSB15248-9	P40
RSB15248-10	EVplus
RSB15248-11	EVplus
RSB15248-12	EVminus
RSB15248-13	EVminus
RSB15248-14	EVminus
RSB15248-15	EVplus
RSB15248-16	Myc
RSB15248-17	Myc
RSB15248-18	Myc
```
Then, we normalized read counts.
```R
# import data
dat <- read.table("counts.bgh-milRNA.txt", header=TRUE, sep=" ")
names(dat) <- c("geneID","length",
                "RSB15248-1", "RSB15248-2", "RSB15248-3", "RSB15248-4", "RSB15248-5", "RSB15248-6",
                "RSB15248-7", "RSB15248-8", "RSB15248-9", "RSB15248-10", "RSB15248-11", "RSB15248-12",
                "RSB15248-13", "RSB15248-14", "RSB15248-15", "RSB15248-16", "RSB15248-17", "RSB15248-18")
info <- read.table("conditions.txt", header = T, sep = '\t')
# normalize read counts
dds <- DESeqDataSetFromMatrix(dat, info, ~condition) 
keep <- rowSums(counts(dds)) >= 10 # filter out non-expressors
dds <- dds[keep,]
# choose a reference level(which level represents the control group or to compare with)
dds$condition <- factor(dds$condition, levels = c("Epi","Myc","Hau","P40","EVplus","EVminus"))
DESeqSizeFactors <- estimateSizeFactors(dds)
sizeFactors(DESeqSizeFactors)
colData(DESeqSizeFactors)
normCounts <- counts(DESeqSizeFactors, normalized = T)
colSums(normCounts) # should be equal across all samples after normalizing
lognormCounts <- log2(normCounts + 1) # log transformation
# to visually inspect noralization
boxplot(dat,notch=F,main="raw counts",ylab="read counts")
boxplot(normCounts,notch=F,main="normalized",ylab="read counts")
```

### Differential expression analysis and LFC shrinkage
We used `DESeq` to calculate log fold change and p-values of comparisons. 
```R
dds <- DESeq(dds)
resultsNames(dds) # to recal the names of the contrasts
DGEresults.myc <- results(dds, contrast=c("condition","Myc","Epi"), alpha=0.05)
DGEresults.hau <- results(dds, contrast=c("condition","Hau","Epi"), alpha=0.05)
DGEresults.p40 <- results(dds, contrast=c("condition","P40","Epi"), alpha=0.05)
summary(DGEresults.hau)

## 
## out of 1168 with nonzero total read count
## adjusted p-value < 0.05
## LFC > 0 (up)       : 59, 5.1%
## LFC < 0 (down)     : 53, 4.5%
## outliers [1]       : 7, 0.6%
## low counts [2]     : 656, 56%
## (mean count < 3)
```
LFC shrinkage by false discovery rate calculation (Benjamini-Hochberg) was done with `apeglm`. Here shown for Hau vs. Epi, all steps were repeated for Myc vs. Epi and P40 vs. Epi as well. 
```R
resLFC <- lfcShrink(dds, coef="Hau", type="apeglm")
## using 'apeglm' for LFC shrinkage. If used in published research, please cite:
##     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
##     sequence count data: removing the noise and preserving large differences.
##     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
resLFC
table(resLFC$padj < 0.1)
## 
## FALSE  TRUE 
##   384   143
resIHW <- results(dds,filterFun=ihw)
## Only 1 bin; IHW reduces to Benjamini Hochberg (uniform weights)
summary(resIHW)
## 
## out of 1168 with nonzero total read count
## adjusted p-value < 0.1
## LFC > 0 (up)       : 59, 5.1%
## LFC < 0 (down)     : 51, 4.4%
## outliers [1]       : 7, 0.6%
## [1] see 'cooksCutoff' argument of ?results
## see metadata(res)$ihwResult on hypothesis weighting
sum(resIHW$padj < 0.1, na.rm=TRUE)
## [1] 110
metadata(resIHW)$ihwResult
## ihwResult object with 1168 hypothesis tests 
## Nominal FDR control level: 0.1 
## Split into 1 bins, based on an ordinal covariate
```
For writing results to tables, here for Hau vs. Epi. 
```R
DGEresultsSorted <- resLFC[order(resLFC$padj),]
DGEgenes <- rownames(subset(DGEresultsSorted, padj < 0.1))
table <- subset(DGEresultsSorted, padj < 0.1)
write.csv(table, "DESEQ_Results_Hau-vs-Epi.csv")
hm.mat_DGEgenes <- lognormCounts[DGEgenes, ]
write.csv(hm.mat_DGEgenes,"DESEQ_Counts_Hau-vs-Epi.csv")
```
To generate a heatmap of differentially expressed genes:
```R
aheatmap(hm.mat_DGEgenes, Rowv = TRUE, Colv = TRUE, distfun = "euclidean", hclustfun = "median")
```

## Differential expression analysis with `EdgeR`
Loading required packages. 
```R
require(limma)
require(edgeR)
require(reshape2)
require(ggplot2)
require(NMF)
```

### Read count normalization
Importing, filtering, and normalizing data. 
```R
# import data
dat <- read.table("counts.bgh-milRNA.txt", header=TRUE, sep=" ")
names(dat) <- c("geneID","length",
                "RSB15248-1", "RSB15248-2", "RSB15248-3", "RSB15248-4", "RSB15248-5", "RSB15248-6",
                "RSB15248-7", "RSB15248-8", "RSB15248-9", "RSB15248-10", "RSB15248-11", "RSB15248-12",
                "RSB15248-13", "RSB15248-14", "RSB15248-15", "RSB15248-16", "RSB15248-17", "RSB15248-18")
sample.edger <- factor(c(rep("Hau",3), rep("Epi",3)))
sample.edger <- relevel(sample.edger, ref = "Epi")
edgeRList <- DGEList(counts = dat, group = sample.edger)
#filter data by cpm
keep <- rowSums(cpm(edgeRList) >= 1) >= 3
edgeRList <- edgeRList[keep, ]
edgeRList$samples$lib.size <- colSums(edgeRList$counts)
head(edgeRList$samples)
##       group lib.size norm.factors
## epi02   Epi    54166            1
## epi03   Epi    46363            1
## epi04   Epi    43277            1
## hau02   Hau     2544            1
## hau03   Hau     2120            1
## hau04   Hau     2739            1
```

### Differential expression analysis
Expression analysis via `EdgeR` was done as follows. 
```R
edgeRList <- calcNormFactors(edgeRList , method = "TMM")
design <- model.matrix(~sample.edger) # sample.edger defined above. Redefine for other comparisons
edgeRList <- estimateDisp(edgeRList, design)
edger_fit <- glmFit(edgeRList, design)
edger_lrt <- glmLRT(edger_fit)
# Filtering for significantly differentially expressed genes. FDR is defined by adjust.method. 
DGE.results_edgeR <- topTags(edger_lrt, n = Inf, p.value = 0.05, sort.by = "PValue", adjust.method = "BH")
table <- DGE.results_edgeR$table
write.csv(table, "EdgeR_Results_Hau-vs-Epi.csv")
```

## Differential expression analysis with `Limma-VOOM`
Loading required packages. 
```R
require(limma)
require(edgeR)
require(reshape2)
require(ggplot2)
require(NMF)
require(pheatmap)
```

### Read count normalization
Importing, filtering, and normalizing read counts for `limma`.
```R
# import data
counts <- read.table("counts.bgh-milRNA.txt", header=TRUE, sep=" ")
names(counts) <- c("geneID","length",
                "RSB15248-1", "RSB15248-2", "RSB15248-3", "RSB15248-4", "RSB15248-5", "RSB15248-6",
                "RSB15248-7", "RSB15248-8", "RSB15248-9", "RSB15248-10", "RSB15248-11", "RSB15248-12",
                "RSB15248-13", "RSB15248-14", "RSB15248-15", "RSB15248-16", "RSB15248-17", "RSB15248-18")
info <- read.table("conditions.txt", header = T, sep = '\t')
# convert count table to matrix
d0 <- DGEList(counts)
d0 <- calcNormFactors(d0)
cutoff <- 1
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,]
# defining samples
count.names <- colnames(counts)
sample <- substr(count.names, 1, nchar(count.names) - 2) 
condition <- substr(count.names, nchar(count.names) - 1, nchar(count.names) - 1)
# grouping data
group <- interaction(info$sample, info$condition)
mm <- model.matrix(~0 + group)
y <- voom(d, mm, plot = T)
```

### Differential expression analysis
Making contrasts to test pairs. 
```R
fit <- lmFit(y, mm)
contr <- makeContrasts(
    Hau = groupHau - groupEpi, 
    Myc = groupMyc - groupEpi,
    P40 = groupP40 - groupEpi,
    EVplus = groupEVplus - groupEpi,
    levels = colnames(coef(fit)))
contr
##             Contrasts
## Levels       groupEpi - groupHau
##   groupEpi                     1
##   groupHau                    -1
```
Running `limma` statistics to identify significantly differentially expressed genes. 
```R
tmp <- contrasts.fit(fit, contr)
bayes <- eBayes(tmp)
tfit <- treat(fit, lfc=1)
dt1 <- decideTests(tfit,p.value=0.05,adjust.method="fdr")
```
Inspecting results.
```R
head(dt1)
## TestResults matrix
##                   groupEpi   groupHau
## BghMilRNA_00001          1          1
## BghMilRNA_00002          1          1
## BghMilRNA_00003          1          1
## BghMilRNA_00004          1          1
## BghMilRNA_00006          1          1
## BghMilRNA_00007          1          1
summary(decideTests(bayes))
##            groupEpi - groupHau
## Down                       533
## NotSig                    1630
## Up                         230
summary(dt1)
##          groupEpi   groupHau
## Down            0          0
## NotSig          0          1
## Up           2393       2392
top.table <- topTable(bayes, sort.by = "P", n = Inf, p.value=0.05)
write.csv(top.table,"Limma_Results.csv")
# for each sample separately
HauEpi <- topTreat(tfit, coef=1, n=Inf)
MycEpi <- topTreat(tfit, coef=2, n=Inf)
P40Epi <- topTreat(tfit, coef=3, n=Inf)
EVplusEpi <- topTreat(tfit, coef=4, n=Inf)
write.csv(HauEpi,"Limma_Results_Hau-vs-Epi.csv")
write.csv(MycEpi,"Limma_Results_Myc-vs-Epi.csv")
write.csv(P40Epi,"Limma_Results_P40-vs-Epi.csv")
write.csv(EVplusEpi,"Limma_Results_EVplus-vs-Epi.csv")
```
